src/Symfony/Contracts/HttpClient/Test/Fixtures/web/index.php

```diff
@@ -63,6 +63,10 @@
         header('Content-Type: application/json', true, 404);
         break;
 
+    case '/404/not-json':
```

Comment:
Can we reuse an existing endpoint instead? Because we now have a cross package dependency: the new version of contracts is needed to run the new test case. But I'd prefer not bumping deps in here.

Reply:
@nicolas-grekas Like I mentioned in the test, if the content-type is json `getContent` gets called and the test doesn't fail when it should.

Reply:
Sorry for asking. Could we successfully test this in any other way without modifying `src/Symfony/contracts/HttpClient/*`?

Reply:
@Nyholm I'll take another look later, maybe there's something I've missed.

Reply:
Done, 304 work as well.

Reply:
Awesome. Thank you